apis/logging/v1/filter_types.go

```diff
@@ -0,0 +1,13 @@
+package v1
+
+// NOTE: The Enum validation on FilterSpec.Type must be updated if the list of
+// known types changes.
+
+// FilterTypeSpec is a union of filter specification types.
+// The fields of this struct define the set of known filter types.
+type FilterTypeSpec struct {
```

Comment:
We seem to be still missing "Type" to point us to which union is the one to review.  I see we have logic to check that only one is filled but that seems unnecessary if the spec tells us the type

Reply:
Fixed. Only the *target* tests are failing now, and the failures don't seem related to these changes.
Are those tests stable?